hw/ip/otbn/util/shared/decode.py

```diff
@@ -31,13 +31,32 @@ def __init__(self, symbols: Dict[str, int], insns: Dict[int, int],
                 raise ValueError(
                     'No legal decoding for mnemonic: {}'.format(mnem))
             insn = INSNS_FILE.mnemonic_to_insn.get(mnem)
+            if insn is None:
+                raise ValueError(
+                    'No instruction found for mnemonic: {}'.format(mnem))
+            if insn.encoding is None:
+                raise ValueError(
+                    'No encoding found for instruction: {}'.format(mnem))
```

Comment:
Again, this can be an assertion: we will never decide that an word disassembles to a given instruction unless that instruction has an encoding.